ibis/pandas/execution/selection.py

```diff
@@ -231,10 +259,11 @@ def _compute_predicates(table_op, predicates, data, scope, **kwargs):
                 new_data = data.loc[:, mapping.keys()].rename(columns=mapping)
             else:
                 new_data = data
-            additional_scope[root_table] = new_data
+            item = make_scope_item(root_table, new_data, timecontext)
+            additional_scope = toolz.merge(additional_scope, item)
 
-        new_scope = toolz.merge(scope, additional_scope)
-        yield execute(predicate, scope=new_scope, **kwargs)
+        set_scope_item(scope, additional_scope)
```

Comment:
I am confused. This seems to be `merge_scope` instead of `set_scope_item`

Is the second param also a "scope"?

Reply:
Yes the second param is a dict of `scope_item`s that could be added to scope.
`set_scope_item` actually could accept another `scope` as well since it tries to iterate key-value pairs.
I agree it's a little confusing if we want to distinguish `merge`(set multiple items) and `set` (set singe item).